x-pack/plugins/monitoring/server/alerts/missing_data_alert.ts

```diff
@@ -0,0 +1,526 @@
+}
+
+export class MissingDataAlert extends BaseAlert {
+  public static paramDetails = {
+    duration: {
+      label: i18n.translate('xpack.monitoring.alerts.missingData.paramDetails.duration.label', {
+        defaultMessage: `Notify if data is missing for`,
+      }),
+      type: AlertParamType.Duration,
+    } as CommonAlertParamDetail,
+    limit: {
+      label: i18n.translate('xpack.monitoring.alerts.missingData.paramDetails.limit.label', {
+        defaultMessage: `Look this far back in time for any data`,
+      }),
+      type: AlertParamType.Duration,
+    } as CommonAlertParamDetail,
+  };
+
+  public type = ALERT_MISSING_DATA;
+  public label = i18n.translate('xpack.monitoring.alerts.missingData.label', {
+    defaultMessage: 'Missing data',
+  });
+
+  protected defaultParams: MissingDataParams = {
+    duration: DEFAULT_DURATION,
+    limit: DEFAULT_LIMIT,
+  };
+
+  protected actionVariables = [
+    {
+      name: 'internalShortMessage',
+      description: i18n.translate(
+        'xpack.monitoring.alerts.missingData.actionVariables.internalShortMessage',
+        {
+          defaultMessage: 'The short internal message generated by Elastic.',
+        }
+      ),
+    },
+    {
+      name: 'internalFullMessage',
+      description: i18n.translate(
+        'xpack.monitoring.alerts.missingData.actionVariables.internalFullMessage',
+        {
+          defaultMessage: 'The full internal message generated by Elastic.',
+        }
+      ),
+    },
+    {
+      name: 'state',
+      description: i18n.translate('xpack.monitoring.alerts.missingData.actionVariables.state', {
+        defaultMessage: 'The current state of the alert.',
+      }),
+    },
+    {
+      name: 'stackProducts',
+      description: i18n.translate(
+        'xpack.monitoring.alerts.missingData.actionVariables.stackProducts',
+        {
+          defaultMessage: 'The stack products missing data.',
+        }
+      ),
+    },
+    {
+      name: 'count',
+      description: i18n.translate('xpack.monitoring.alerts.missingData.actionVariables.count', {
+        defaultMessage: 'The number of stack products missing data.',
+      }),
+    },
+    {
+      name: 'clusterName',
+      description: i18n.translate(
+        'xpack.monitoring.alerts.missingData.actionVariables.clusterName',
+        {
+          defaultMessage: 'The cluster to which the stack products belong.',
+        }
+      ),
+    },
+    {
+      name: 'action',
+      description: i18n.translate('xpack.monitoring.alerts.missingData.actionVariables.action', {
+        defaultMessage: 'The recommended action for this alert.',
+      }),
+    },
+    {
+      name: 'actionPlain',
+      description: i18n.translate(
+        'xpack.monitoring.alerts.missingData.actionVariables.actionPlain',
+        {
+          defaultMessage: 'The recommended action for this alert, without any markdown.',
+        }
+      ),
+    },
+  ];
+
+  protected async fetchData(
+    params: CommonAlertParams,
```

Comment:
You can also set the type as `params: MissingDataParams` here, and `params: CommonAlertParams | unknown` in `base_alerts.ts` That way you won't need to do any funky casting/recasting